{
  "configurations": {
    "CVE_data_version": "4.0",
    "nodes": [
      {
        "cpe_match": [
          {
            "cpe23Uri": "cpe:2.3:a:storage_project:storage:*:*:*:*:*:*:*:*",
            "versionEndExcluding": "1.28.1",
            "vulnerable": true
          }
        ],
        "operator": "OR"
      },
      {
        "cpe_match": [
          {
            "cpe23Uri": "cpe:2.3:a:redhat:openshift_container_platform:4.0:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:o:redhat:enterprise_linux:8.0:*:*:*:*:*:*:*",
            "vulnerable": true
          }
        ],
        "operator": "OR"
      }
    ]
  },
  "cve": {
    "CVE_data_meta": {
      "ASSIGNER": "cve@mitre.org",
      "ID": "CVE-2021-20291"
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "A deadlock vulnerability was found in 'github.com/containers/storage' in versions before 1.28.1. When a container image is processed, each layer is unpacked using `tar`. If one of those layers is not a valid `tar` archive this causes an error leading to an unexpected situation where the code indefinitely waits for the tar unpacked stream, which never finishes. An attacker could use this vulnerability to craft a malicious image, which when downloaded and stored by an application using containers/storage, would then cause a deadlock leading to a Denial of Service (DoS)."
        }
      ]
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": [
            {
              "lang": "en",
              "value": "CWE-667"
            }
          ]
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "name": "https://bugzilla.redhat.com/show_bug.cgi?id=1939485",
          "refsource": "MISC",
          "tags": [
            "Issue Tracking",
            "Patch",
            "Third Party Advisory"
          ],
          "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1939485"
        }
      ]
    }
  },
  "impact": {
    "baseMetricV2": {
      "acInsufInfo": false,
      "cvssV2": {
        "accessComplexity": "MEDIUM",
        "accessVector": "NETWORK",
        "authentication": "NONE",
        "availabilityImpact": "COMPLETE",
        "baseScore": 7.1,
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "vectorString": "AV:N/AC:M/Au:N/C:N/I:N/A:C",
        "version": "2.0"
      },
      "exploitabilityScore": 8.6,
      "impactScore": 6.9,
      "obtainAllPrivilege": false,
      "obtainOtherPrivilege": false,
      "obtainUserPrivilege": false,
      "severity": "HIGH",
      "userInteractionRequired": true
    },
    "baseMetricV3": {
      "cvssV3": {
        "attackComplexity": "LOW",
        "attackVector": "NETWORK",
        "availabilityImpact": "HIGH",
        "baseScore": 6.5,
        "baseSeverity": "MEDIUM",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "REQUIRED",
        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
        "version": "3.1"
      },
      "exploitabilityScore": 2.8,
      "impactScore": 3.6
    }
  },
  "lastModifiedDate": "2021-04-06T17:06Z",
  "publishedDate": "2021-04-01T18:15Z"
}